### PR TITLE
Update Readme, bump nodejs version to current LTS.

### LINKED ISSUE #133: Clarify Nodejs version
The Readme says "NodeJS v10+" but the .nvmrc file and  .circleci/config.yml  say Node version 8.x.

### PR DESCRIPTION
**Pull request by guidorice**
- change `.nvmrc` from v8 to v12.16 which is the current LTS version
  of nodejs.

- update version of nodejs in Readme.

- fix markdown lint warnings in Readme.

- restructure Readme to avoid setup error.

- use yarn instead of npm in Readme.

Closes #133

### CODE REVIEW

**Comment by guidorice** [comment]
@kamicut when you review can you confirm about where the `yarn run migrate` command belongs? The `yarn install` step was too far down so I moved it up before where the javascript dependencies are needed.

File: README.md [hunk 1]
```diff
@@ -1,43 +1,59 @@
...
-```
-docker-compose run --rm hydra migrate sql --yes postgres://postgres@host.docker.internal/osm-teams?sslmode=disable
-```
+    For the rest of this documentation, we will assume that the database location is `postgres://postgres@localhost/osm-teams?sslmode=disable` on your local machine. Inside docker, that location is `postgres://postgres@host.docker.internal/osm-teams?sslmode=disable`
 
-4. Build the images:
-```
-docker-compose build
-```
+1. Create a a `.env` file by copying `.env.sample` and replacing the values as needed. `OSM_CONSUMER_KEY` and `OSM_CONSUMER_SECRET` are values obtained by creating a new OAuth app on openstreetmap.org. The .env file can contain:
```
Comment (kamicut) [inline README.md]:
```suggestion
2. Create a a `.env` file by copying `.env.sample` and replacing the values as needed. `OSM_CONSUMER_KEY` and `OSM_CONSUMER_SECRET` are values obtained by creating a new OAuth app on openstreetmap.org. The .env file can contain:
```

File: README.md [hunk 2]
```diff
@@ -1,43 +1,59 @@
...
-docker-compose -f docker-compose.dev.yml up
-```
+    ```bash
+    OSM_CONSUMER_KEY=<osm-teams-app>
+    OSM_CONSUMER_SECRET=<osm-teams-app-secret>
+    DSN=postgres://postgres@host.docker.internal/osm-teams?sslmode=disable
+    SECRETS_SYSTEM=<random-guid>
+    ```
+
+1. Install Javascript dependencies
```
Comment (kamicut) [inline README.md]:
```suggestion
3. Install Javascript dependencies
```

File: README.md [hunk 3]
```diff
@@ -1,43 +1,59 @@
...
+
+    ```bash
+    yarn install
+    yarn run migrate
+    ```
+
+1. Migrate the database
```
Comment (kamicut) [inline README.md]:
```suggestion
4. Migrate the database
```

File: README.md [hunk 4]
```diff
@@ -1,43 +1,59 @@
...
+
+    ```bash
+    docker-compose run --rm hydra migrate sql --yes postgres://postgres@host.docker.internal/osm-teams?sslmode=disable
+    ```
+
+1. Build the images:
```
Comment (kamicut) [inline README.md]:
```suggestion
5. Build the images:
```

File: README.md [hunk 5]
```diff
@@ -1,43 +1,59 @@
...
+
+    ```bash
+    docker-compose build
+    ```
+
+1. Start Hydra and the server
```
Comment (kamicut) [inline README.md]:
```suggestion
6. Start Hydra and the server
```

**Comment by guidorice** [comment]
@kamicut this is the reason for using list auto-numbering:

- it's a pain to renumber all the items if you edit or insert list items
- github and stackoverflow, and other markdown flavors all support this syntax

https://stackoverflow.com/questions/12145232/create-an-automatically-numbered-list#12145270

**Pull request merged**